src/render/ui/font.rs

```diff
@@ -0,0 +1,135 @@
+use crate::render::texture::{Texture, TextureType};
+use crate::render::ui::Rect;
+
+use math::utils::next_power_of_two;
+use rusttype::{Font as FontType, Point, Scale};
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+use std::str::Chars;
+
+pub struct DrawableFontCharacter {
+    texture: Texture,
+    width: u32,
+    height: u32,
+    top_bearing: i32,
+}
+
+pub struct FontCharacter {
+    drawable: Option<DrawableFontCharacter>,
+    side_bearing: i32,
+    advance: f32,
+}
+
+pub struct Font {
+    chars: HashMap<char, FontCharacter>,
+}
+
+impl Font {
+    pub fn new(path: &Path, size: f32) -> Option<Self> {
+        let data = fs::read(path.to_str()?).ok()?;
+        let font = FontType::try_from_bytes(data.as_slice())?;
+
+        let scale = Scale::uniform(size);
+
+        let mut chars = HashMap::new();
```

Comment:
nit: maybe declare it immutable by mapping some `u8 -> (u8, Option<FontCharacter>)` fn to the range and collecting it in a HashMap?